### PR TITLE
Добавление контекста

### PR DESCRIPTION
**Pull request by NikitaGoroshko7**
https://trello.com/c/BRRCrdij/3-%D1%80%D0%B5%D0%B3%D0%B8%D1%81%D1%82%D1%80%D0%B0%D1%86%D0%B8%D1%8F-%D0%B8-%D0%B0%D0%B2%D1%82%D0%BE%D1%80%D0%B8%D0%B7%D0%B0%D1%86%D0%B8%D1%8F
Has been changed ApplicationContext
Has been added validation for models
Has been added CustomValidation
Has been added some new services in file startap and usings in file import.cs

### CODE REVIEW

**Review by GalayM — COMMENTED** [review]
Need to use Identity context

File: CrowdFundingApplication/Controllers/AccountController.cs [hunk 1]
```diff
@@ -1,14 +1,63 @@
...
+                var result = await _userManager.CreateAsync(user, model.Password);
+
+                if (result.Succeeded)
+                {
+                    await _signInManager.SignInAsync(user, false);
+                    return RedirectToAction("MainPage", "Home");
+                }
+                else
+                    foreach (var error in result.Errors)
+                        ModelState.AddModelError(string.Empty, error.Description);
```
Comment (GalayM) [inline CrowdFundingApplication/Controllers/AccountController.cs:34]: Always use {}

File: CrowdFundingApplication/Controllers/AccountController.cs [hunk 2]
```diff
@@ -1,14 +1,63 @@
...
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Login(InputViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                var result = await _signInManager.PasswordSignInAsync(model.Name, model.Password, model.RememberMe, false);
+
+                if (result.Succeeded)
+                    return RedirectToAction("MainPage", "Home");
```
Comment (GalayM) [inline CrowdFundingApplication/Controllers/AccountController.cs:55]: Always use {}

File: CrowdFundingApplication/Controllers/AccountController.cs [hunk 3]
```diff
@@ -1,14 +1,63 @@
...
+                else
+                    ModelState.AddModelError("", "Неправильный логин и (или) пароль");
```
Comment (GalayM) [inline CrowdFundingApplication/Controllers/AccountController.cs:59]: Always use {}

File: CrowdFundingApplication/Controllers/HomeController.cs [hunk 4]
```diff
@@ -18,16 +9,25 @@ public HomeController(ILogger<HomeController> logger)
...
+        [Authorize]
         public IActionResult Privacy()
         {
             return View();
         }
 
+        [Authorize]
+        public IActionResult MainPage()
+        {
+
```
Comment (GalayM) [inline CrowdFundingApplication/Controllers/HomeController.cs]: Why is there empty space?

File: CrowdFundingApplication/Data/ApplicationContext.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿namespace CrowdFundingApplication.Data;
+
+public class ApplicationContext : IdentityDbContext<User>
+{
+    public ApplicationContext(DbContextOptions<ApplicationContext> options) : base(options) 
+    {
+        Database.EnsureCreated();
+    }
+
```
Comment (GalayM) [inline CrowdFundingApplication/Data/ApplicationContext.cs]: ???

File: CrowdFundingApplication/Models/InputViewModel.cs [hunk 6]
```diff
@@ -2,10 +2,14 @@
...
-        public string Email { get; set; }
+        [Required(ErrorMessage = "Вы не ввели имя!")]
+        public string Name { get; set; }
 
+        [Required(ErrorMessage = "Вы не ввели пароль!")]
+        [DataType(DataType.Password)]
         public string Password { get; set; }
 
         public bool RememberMe { get; set; }
+
```
Comment (GalayM) [inline CrowdFundingApplication/Models/InputViewModel.cs]: ???

File: CrowdFundingApplication/Validation/CustomValidation.cs [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+            List<IdentityError> errors = new List<IdentityError>();
+
+            if (user.UserName.Contains("admin"))
+            {
+                errors.Add(new IdentityError
+                {
+                    Description = "Недопустимое имя пользоваля!"
+                });
+            }
+            return Task.FromResult(errors.Count == 0 ?
```
Comment (GalayM) [inline CrowdFundingApplication/Validation/CustomValidation.cs]: Any()

File: CrowdFundingApplication/Controllers/AccountController.cs (hunk 1 above)
Comment (NikitaGoroshko7) [reply CrowdFundingApplication/Controllers/AccountController.cs:34]: Has been added {}

File: CrowdFundingApplication/Controllers/AccountController.cs (hunk 2 above)
Comment (NikitaGoroshko7) [reply CrowdFundingApplication/Controllers/AccountController.cs:55]: Has been added {}

File: CrowdFundingApplication/Controllers/AccountController.cs (hunk 3 above)
Comment (NikitaGoroshko7) [reply CrowdFundingApplication/Controllers/AccountController.cs:59]: Has been added {}

File: CrowdFundingApplication/Controllers/HomeController.cs (hunk 4 above)
Comment (NikitaGoroshko7) [reply CrowdFundingApplication/Controllers/HomeController.cs]: Has been deleted the empty space

File: CrowdFundingApplication/Data/ApplicationContext.cs (hunk 5 above)
Comment (NikitaGoroshko7) [reply CrowdFundingApplication/Data/ApplicationContext.cs]: Has been deleted empty line

File: CrowdFundingApplication/Models/InputViewModel.cs (hunk 6 above)
Comment (NikitaGoroshko7) [reply CrowdFundingApplication/Models/InputViewModel.cs]: Has been deleted empty line

File: CrowdFundingApplication/Validation/CustomValidation.cs (hunk 7 above)
Comment (NikitaGoroshko7) [reply CrowdFundingApplication/Validation/CustomValidation.cs]: Has been added Any() instend Count == 0

**Review by GalayM — APPROVED** [review]

**Pull request merged**